### PR TITLE
#5300 - Fixed

### PR DESCRIPTION
**Pull request by nikhilbarar**
This pull request includes a

[5300] Bug fix
The following changes were made

Trimmed the term for default and matcher

Fixed Issue #5300

### CODE REVIEW

**Comment by pedrofurtado** [comment]
@nikhilbarar Please, remove all files in this PR from `dist` folder. They are automatically generated during build process.

File: src/js/select2/compat/matcher.js [hunk 1]
```diff
@@ -8,6 +8,9 @@ define([
       if (params.term == null || $.trim(params.term) === '') {
         return match;
       }
+      else {
```
Comment (hakan0xFF) [inline src/js/select2/compat/matcher.js]: the `params.term` is trimmed using a jQuery function so you could do

```js
params.term = $.trim(params.term)
if (params.term === '') {
  return match;
}
```

there is no need of an `else`

File: src/js/select2/defaults.js [hunk 2]
```diff
@@ -311,6 +311,9 @@ define([
       if ($.trim(params.term) === '') {
         return data;
       }
+      else {
```
Comment (hakan0xFF) [inline src/js/select2/defaults.js]: see comment above

File: src/js/select2/compat/matcher.js (hunk 1 above)
Comment (pedrofurtado) [reply src/js/select2/compat/matcher.js]: @hakan0xFF But we have to be aware if `$.trim` works well when parameter is `null`. The result of `$.trim` is always a string, even blank? Or is there some case that it returns null? This should be included in the code.

Comment (hakan0xFF) [reply src/js/select2/compat/matcher.js]: if the value/parameter is null the trim function returns an empty string. so it returns always a string. see https://github.com/select2/select2/blob/develop/vendor/jquery-2.1.0.js#L402

**Pull request closed without merging**